build-and-deploy.sh

```diff
@@ -28,7 +28,13 @@ if [ "${TRAVIS_BRANCH}" = "master" ] && [ "${TRAVIS_PULL_REQUEST}" = "false" ];
         # because we can infer it based on GitHub Pages conventions
         if [ "${BASE_URL}" = "" ]; then
             BASE_URL=$(echo $TRAVIS_REPO_SLUG | sed -e 's@.*/@/@')
+            FULL_URL="https://"$(echo $TRAVIS_REPO_SLUG | sed 's@/.*@.github.io@')
+        else
+            # Everything is shoved into BASE_URL so this can be empty
+            FULL_URL=
         fi
+        # FULL_URL is for the opengraph tags, which can't be relative
+        perl -pe 's@^url:.*@url: '"${FULL_URL}"'@' -i _config.yml
```

Comment:
How about using override config file instead of rewrite the original config file?

```shell
    custom_config_yml="_config.override.yml"
    touch ${custom_config_yml}
    if [ "${TRAVIS_REPO_SLUG}" = "apache/arrow-site" ]; then
        # Production
        TARGET_BRANCH=asf-site
        BASE_URL=
    else
        # On a fork, so we'll deploy to GitHub Pages
        TARGET_BRANCH=gh-pages
        # You could supply an alternate BASE_URL, but that's not necessary
        # because we can infer it based on GitHub Pages conventions
        if [ "${BASE_URL}" = "" ]; then
            BASE_URL=$(echo $TRAVIS_REPO_SLUG | sed -e 's@.*/@/@')
            echo "url: https://$(echo $TRAVIS_REPO_SLUG | sed 's@/.*@.github.io@')" >> ${custom_config_yml}
        fi
    fi

    # Build
    JEKYLL_ENV=production \
        bundle exec \
            jekyll build \
               --baseurl="${BASE_URL}" \
               --config=_config.yml,${custom_config_yml}
```

See also: https://jekyllrb.com/docs/configuration/options/#build-command-options

Reply:
I guess, but is that really an improvement? Your proposal may be more pure, and I'm happy to do it if you insist, but IMO it's harder to read and I'm not sure it's worth it. This is a script that only runs on a Travis-CI environment that gets destroyed after use, so we're not hurting anything by altering `_config.yml`.